src/pypromice/process/aws.py

```diff
@@ -91,44 +91,79 @@ def getL1(self):
         logger.info('Level 1 processing...')
         self.L0 = [addBasicMeta(item, self.vars) for item in self.L0]
         self.L1 = [toL1(item, self.vars) for item in self.L0]
-        self.L1A = reduce(xr.Dataset.combine_first, self.L1)
+
+        if self.merge_flag:
+            self.L1A = self.hard_merge(self.L1)
+        else:
+            self.L1A = reduce(xr.Dataset.combine_first, self.L1)
 
     def getL2(self):
         '''Perform L1 to L2 data processing'''
         logger.info('Level 2 processing...')
         self.L2 = toL2(self.L1A, vars_df=self.vars)
+        self.L2 = self.resample(self.L2)
+        self.L2 = reformat_time(self.L2)
+
+        # Switch gps_lon to negative (degrees_east)
+        # Do this here, and NOT in addMeta, otherwise we switch back to positive
+        # when calling getMeta in joinL2! PJW
+        if self.L2.attrs['station_id'] not in ['UWN', 'Roof_GEUS', 'Roof_PROMICE']:
+            self.L2['gps_lon'] = self.L2['gps_lon'] * -1
+
+        # Add variable attributes and metadata
+        self.L2 = self.addAttributes(self.L2)
+
+        # Round all values to specified decimals places
+        self.L2 = roundValues(self.L2, self.vars)
 
     def getL3(self):
         '''Perform L2 to L3 data processing, including resampling and metadata
         and attribute population'''
         logger.info('Level 3 processing...')
         self.L3 = toL3(self.L2)
 
-        # Resample L3 product
+    def resample(self, dataset):       
+        '''Resample dataset to specific temporal resolution (based on input
+        data type)'''
         f = [l.attrs['format'] for l in self.L0]
         if 'raw' in f or 'STM' in f:
             logger.info('Resampling to 10 minute')
-            self.L3 = resampleL3(self.L3, '10min')
+            resampled = resampleL2(dataset, '10min')
         else:
-            self.L3 = resampleL3(self.L3, '60min')
+            resampled = resampleL2(dataset, '60min')
             logger.info('Resampling to hour')
-
-        # Re-format time
-        t = self.L3['time'].values
-        self.L3['time'] = list(t)
-
-        # Switch gps_lon to negative (degrees_east)
-        # Do this here, and NOT in addMeta, otherwise we switch back to positive
-        # when calling getMeta in joinL3! PJW
-        if self.L3.attrs['station_id'] not in ['UWN', 'Roof_GEUS', 'Roof_PROMICE']:
-            self.L3['gps_lon'] = self.L3['gps_lon'] * -1
-
-        # Add variable attributes and metadata
-        self.L3 = self.addAttributes(self.L3)
-
-        # Round all values to specified decimals places
-        self.L3 = roundValues(self.L3, self.vars)
```

Comment:
I'd be great to hear from you whether you need values from L2 to be rounded before you use them in the BUFR file processing. If not, then we could do rounding at the end of the pipeline, after join_l3.

The same question goes to resampling, with the difference that I would like (for now) to keep the 10_min data only a `L2` or `L2_merge` product option. I don't think it is wise to use ressources and processing time to compute turbulent fluxes, smoothed coordinates... for a 10_minute product that is not yet posted. Data users who have been using 10 minute data usually want to do these calculation themselves. So I'd recommend to have the hourly `resample` being applied after `join_l2` so that `l2tol3` runs on hourly values only, and we can have daily and monthly resample on the `L3` product either before or after the site-wise merge. 

@PennyHow any opinion on this?